### PR TITLE
Select component

### PR DESCRIPTION
**Pull request by zikowang**
Select Component:

AddVocab Screen:
image

AddLanguagePackage Screen:
image
image
image

### CODE REVIEW

File: src/Components/Select/Select.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useEffect, useState, useRef } from "react";
+import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
+
+import "./Select.scss";
+
+const Select = ({ onChange = () => null, required = false, options = [], label = "", value, children, ...props }) => {
```
Comment (luwol03) [inline src/Components/Select/Select.jsx:6]: `children` is unused

File: src/Components/Select/Select.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <span className={`select-label ${showOptions || !!defaultText ? "flow" : ""}`}>{`${label}${
+          required ? " *" : ""
+        }`}</span>
+        {defaultText}
+        <ArrowDropDownIcon className="select-icon" />
+      </div>
+      {showOptions && (
+        <ul className="select-options-list">
+          {options.map((option) => (
+            <li className="select-option" data-name={option.name} key={option.id} onClick={handleOptionClick}>
```
Comment (luwol03) [inline src/Components/Select/Select.jsx:72]: could you use for the key the second parameter of the map method?

**Review by luwol03 — CHANGES_REQUESTED** [review]
could you make it possible to use a translation for the display name but an extra string for the internal name?

**Comment by zikowang** [comment]
@luwol03 would you please updated package-lock for me? :)

File: src/i18n/locales/en.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "global": {
+    "name": "Name",
+    "noOptionMessage": "no options",
```
[the remark this answers is not in the document]
Comment (zikowang) [reply src/i18n/locales/en.json:4]: oh yes

**Review by luwol03 — APPROVED** [review]

**Comment by luwol03** [comment]
Can we merge this beautiful pull request? The text fields look much better after the last change.

**Comment by zikowang** [comment]
> @zikowang Can we merge this beautiful pull request? The text fields look much better after the last change.

Thanks. Let's merge it.

**Pull request merged**